fdbserver/DataDistribution.actor.cpp

```diff
@@ -1418,10 +1467,12 @@ struct DDTeamCollection : ReferenceCounted<DDTeamCollection> {
 				ASSERT_WE_THINK(isMachineTeamHealthy(machineIDs));
 
 				std::sort(machineIDs.begin(), machineIDs.end());
-				if (machineTeamExists(machineIDs)) {
+				int overlap = overlappingMachineMembers(machineIDs);
+				if (overlap == machineIDs.size()) {
 					maxAttempts += 1;
 					continue;
 				}
+				score += 10000*overlap;
```

Comment:
I prefer the constant 1000 to be a knob. 
It is also better to add a comment "1000 must be larger than the largest possible team number a server may have."